src/main/java/org/waarp/openr66/protocol/networkhandler/NetworkServerInitializer.java

```diff
@@ -49,23 +49,21 @@ public NetworkServerInitializer(boolean server) {
     @Override
     protected void initChannel(SocketChannel ch) throws Exception {
         final ChannelPipeline pipeline = ch.pipeline();
+        pipeline.addLast(TIMEOUT, new IdleStateHandler(0, 0,
+                    Configuration.configuration.getTIMEOUTCON(),
+                    TimeUnit.MILLISECONDS));
+        // Global limitation
+        pipeline.addLast(LIMITGLOBAL,
+                Configuration.configuration.getGlobalTrafficShapingHandler());
+        // Per channel limitation
+        pipeline.addLast(LIMITCHANNEL,
+                new ChannelTrafficShapingHandler(
+                    Configuration.configuration.getServerChannelWriteLimit(),
+                    Configuration.configuration.getServerChannelReadLimit(),
+                    Configuration.configuration.getDelayLimit()));
         pipeline.addLast("codec", new NetworkPacketCodec());
```

Comment:
Same here: before the codec was set previously to traffic shaping... Maybe this is the right place there ?

Reply:
See above